scripts/Interface/ui/sans_isis/run_selector_widget.py

```diff
@@ -0,0 +1,109 @@
+from __future__ import (absolute_import, division, print_function)
+
+from PyQt4 import QtGui, QtCore
+
+import ui_run_selector_widget
+from PyQt4.QtCore import pyqtSignal
+from mantidqtpython import MantidQt
+
+
+class RunSelectorWidget(QtGui.QWidget, ui_run_selector_widget.Ui_RunSelectorWidget):
+    manageDirectories = pyqtSignal()
+    browse = pyqtSignal()
+    addRuns = pyqtSignal()
+    removeRuns = pyqtSignal()
+    removeAllRuns = pyqtSignal()
+
+    def __init__(self, parent=None):
+        super(RunSelectorWidget, self).__init__(parent)
+        self.setupUi(self)
+        self._connect_signals()
+
+    def setupUi(self, other):
+        ui_run_selector_widget.Ui_RunSelectorWidget.setupUi(self, other)
+        self.runList.setSelectionMode(QtGui.QAbstractItemView.ExtendedSelection)
+
+    def show_file_picker(self, extensions, search_directories):
+        previous_directories = self._previous_directory_settings()
+        default_directory = search_directories[0]
+        directory = self._previous_or_default_directory(previous_directories, default_directory)
+        file_filter = self._filter_for_extensions(extensions)
+        chosen_files = QtGui.QFileDialog.getOpenFileNames(self, "Select files", directory, file_filter)
+        if chosen_files:
```

Comment:
is it worth putting this if into the presenter?

Reply:
The problem with that is that this is Qt Specific.